### ISSUE TITLE
Flushed up to token index xxxxxx, cache file is xxxxxxxxxxxxxx 

### ISSUE THREAD

**Issue by jovegoh** [ChrisNZL/Tallowmere2#1146]
Flushed up to token index xxxxxx, cache file is xxxxxxxxxxxxxx
causing screen freeze for 1s every 1min

I tried clear cache, data, uninstall, restart phone, reinstall still same issue
short sample of freeze in attached video

Device: Nubia Redmagic5g
OS: Android 11
Game version: 0.3.7g
Downloaded from playstore

![Screenshot_2022-12-03-04-03-26-299](https://user-images.githubusercontent.com/36652777/205376642-33dc95ef-91b8-4278-a11e-62682dcb77f5.png)

https://user-images.githubusercontent.com/36652777/205377591-d773a669-c329-471c-abd4-ddc268d17810.mp4

**Comment by ChrisNZL** [comment]
Thanks for the report.

The message `Flushed up to token index` does not exist in my game's scripts, so it's coming from a plugin I am using somewhere...

Searching Google for `Flushed up to token index`, came across these Unity threads:
- [Unity Analytics making my app sluggish?](https://forum.unity.com/threads/unity-analytics-making-my-app-sluggish.1361308/)
- [Unity Analytics (New) Questions](https://forum.unity.com/threads/unity-analytics-new-questions.1244146/)

Sounds like it's an unwanted feature of Unity Analytics / Unity Gaming Services. I require using Unity Gaming Services to make the In-App Purchase system work, but I do not need the Unity Analytics stuff.

One call a Unity employee is suggesting to use is:
```csharp
AnalyticsService.Instance.SetAnalyticsEnabled(false);
```
So I will likely try adding this to my startup code.

I will have to profile and debug and make sure this does not break the IAP system, but I believe this is the culprit and solution. I will try to get this fixed soon.

**Comment by jovegoh** [comment]
Hi, I kinda found out the reason for this flush thing.
I have been using dns.adsguard.com for my phone DNS

 after off the DNS, the flushing stops.
 I enable again, the flush index reset from low number which doesnt seems to hang the game.

**Comment by jovegoh** [comment]
Hi, new findings: the flush is probably caused by lost connection to whatever server
Reason: due to flush appearing when playing without internet connection (wifi and mobile data off)

**Comment by ChrisNZL** [comment]
Thanks for the further info.

The stuttering and logging should be resolved in upcoming 0.3.7h.

Added the following to the game's `UnityGamingServices` class:
```csharp
// ➡️ Disable Unity Analytics:
// https://forum.unity.com/threads/unity-analytics-making-my-app-sluggish.1361308/
AnalyticsService.Instance.SetAnalyticsEnabled(false); 

// ➡️ Further disabling:
// https://docs.unity3d.com/ScriptReference/Analytics.Analytics-enabled.html
Analytics.enabled = false;
Analytics.deviceStatsEnabled = false;
PerformanceReporting.enabled = false;
```

IAP functionality seems to remain intact.
